src/live_restore/live_restore_fs.c

```diff
@@ -1415,7 +1398,7 @@ __live_restore_fs_atomic_copy_file(WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_FS
 
     /* Open both files and create the temporary destination file. */
     WT_ERR(lr_fs->os_file_system->fs_open_file(
-      lr_fs->os_file_system, wt_session, source_path, type, WT_FS_OPEN_EXCLUSIVE, &source_fh));
```

Comment:
I've dropped `WT_FS_OPEN_EXCLUSIVE` here as it is only relevant when creating new files. 
This flags maps to [`O_EXCL`](https://www.man7.org/linux/man-pages/man2/open.2.html) in our posix layer behaviour, and [`CREATE_ALWAYS`](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-createfilea) for our windows layer.

Neither of these make guarantees that only a single handle is opened on the file. `O_EXCL` is only relevant in conjunction with `O_CREAT` to make sure the file didn't exist previously.
`ALWAYS_CREATE` will actively cause us issues as it truncates the existing file instead of guaranteeing file handle exclusivity like we intended.

Reply:
Would we want `WT_FS_OPEN_READONLY` instead? We only ever want to read from the source and that would be a safety check if there was a bug that tried to write on the source.

Reply:
SGTM. I've added the flag here and to the `fs_open_file` call in `__live_restore_fs_open_in_source`